bundles/org.openhab.binding.bluetooth.bluegiga/src/main/java/org/openhab/binding/bluetooth/bluegiga/internal/BlueGigaTransactionManager.java

```diff
@@ -0,0 +1,371 @@
+        transactionListeners.clear();
+        eventListeners.clear();
+        logger.debug("Closed");
+    }
+
+    private void startTransactionTimer() {
+        transactionTimeoutTimer = executor.schedule(() -> {
+            notifyTransactionTimeout(ongoingTransactionId);
+        }, TRANSACTION_TIMEOUT_PERIOD_MS, TimeUnit.MILLISECONDS);
+    }
+
+    private void cancelTransactionTimer() {
+        if (transactionTimeoutTimer != null) {
+            transactionTimeoutTimer.cancel(true);
+        }
+    }
+
+    private void sendNextFrame() {
+        BlueGigaUniqueCommand nextFrame = sendQueue.poll();
+        if (nextFrame == null) {
+            // Nothing to send
+            logger.trace("Send frame: nothing to send");
+            return;
+        }
+        cancelTransactionTimer();
+        logger.debug("Send frame #{}: {}", nextFrame.getTransactionId(), nextFrame.getMessage());
+        ongoingTransactionId = nextFrame.getTransactionId();
+        serialHandler.sendFrame(nextFrame.getMessage());
+        startTransactionTimer();
+    }
+
+    /**
+     * Add a {@link BlueGigaCommand} frame to the send queue. The sendQueue is a
+     * FIFO queue. This method queues a {@link BlueGigaCommand} frame without
+     * waiting for a response.
+     *
+     * @param transaction
+     *            {@link BlueGigaCommand}
+     */
+    public void queueFrame(BlueGigaUniqueCommand request) {
+        logger.trace("Queue TX BLE frame: {}", request);
+        sendQueue.add(request);
+        logger.trace("TX BLE queue size: {}", sendQueue.size());
+    }
+
+    private void sendNextTransactionIfNoOngoing() {
+        synchronized (this) {
+            logger.trace("Send next transaction if no ongoing");
+            if (ongoingTransactionId == null) {
+                sendNextFrame();
+            }
+        }
+    }
+
+    private void clearOngoingTransactionAndSendNext() {
+        synchronized (this) {
+            logger.trace("Clear ongoing transaction and send next frame from queue");
+            ongoingTransactionId = null;
+            sendNextFrame();
+        }
+    }
+
+    private void addTransactionListener(BluetoothListener<? extends BlueGigaResponse> listener) {
+        if (transactionListeners.contains(listener)) {
+            return;
+        }
+
+        transactionListeners.add(listener);
+    }
+
+    private void removeTransactionListener(BluetoothListener<?> listener) {
+        transactionListeners.remove(listener);
+    }
+
+    /**
+     * Sends an BlueGiga request without waiting for the response.
+     *
+     * @param bleCommand {@link BlueGigaCommand}
+     * @return response {@link Future} {@link BlueGigaResponse}
+     */
+    private <T extends BlueGigaResponse> Future<T> sendBleRequestAsync(final BlueGigaCommand bleCommand,
+            final Class<T> expected) {
+        class TransactionWaiter implements Callable<T>, BluetoothListener<T> {
+            private volatile boolean complete;
```

Comment:
Just to be explicit.
```suggestion
            private volatile boolean complete = false;
```

Reply:
I also like explicit initialisation, but most of the OH maintainers not.